cpp/src/arrow/flight/CMakeLists.txt

```diff
@@ -102,84 +102,21 @@ set(CMAKE_CXX_FLAGS_BACKUP "${CMAKE_CXX_FLAGS}")
 string(REPLACE "/WX" "" CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}")
 string(REPLACE "-Werror " " " CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}")
 
-# Probe the version of gRPC being used to see if it supports disabling server
```

Comment:
@kou Is there a reason we kept the KLUDGE above when removing the `try_compile` calls?

Reply:
No. I should have removed the KLUDGE too in this PR.